### ISSUE TITLE
0.4.3.88 - check_process - check_process and problem with warning and '<'

### ISSUE THREAD

**Issue by garnier-quentin** [mickem/nscp#97]
Hi,

It seems, there is a problem with expressions. I have two notepad opened on my windows. The check returns 2 for the 'count' (good value). But i have a warning if i use 'warning=count<2' ??!! 
```bash
# /usr/lib/nagios/plugins/check_nrpe -H xxxx -p 5666 -t 30 -u -c check_process -a "filter=exe eq 'notepad.exe'" "warning=count<2" "critical="
WARNING: notepad.exe=started|'count'=2;2;0
# /usr/lib/nagios/plugins/check_nrpe -H xxxx -p 5666 -t 30 -u -c check_process -a "filter=exe eq 'notepad.exe'" "warning=count<3" "critical="
WARNING: notepad.exe=started, notepad.exe=started|'count'=2;3;0
# /usr/lib/nagios/plugins/check__nrpe -H xxxx -p 5666 -t 30 -u -c check_process -a "filter=exe eq 'notepad.exe'" "warning=count<1" "critical="
OK: |'count'=2;1;0
```

Thanks.

**Comment by mickem** [comment]
Yes, this is a rather serious logical problem actually...

I will revise this for 0.4.4 but cannot (I think) fix it for 0.4.3 as it requires a major change...

The problem is that the various counts are evaluated through out meaning first match count is 1, second match count is 2 and so on and so forth.
This work as long as you dont check for <...

ie. do we have more then 5 matches works but not less then....


// Michael Medin

**Comment by quentinou** [comment]
Ok. But also it doesn't work with '!=' operator (If i do 'count != 2'). Do you have a workaround to do what i want ?

Thanks for your help.

**Comment by mickem** [comment]
Same thing there since the value changes... you can only doe "more than"...

And unfortunately I am not sure I will be able to help you before 0.4.4 (which is not to far off)... 

I'll think about it...

ON second thought... one somewhat ugly workaround would be some sort of meta check on the performance data?
Could be done with a lua/python script...

I'll try to work something out over the weekend...

// Michael Medin

**Comment by mickem** [comment]
But again, the real fix will be in 0.4.4 then I will move checks till after filtering so we can use the real count values instead of the chaning ones...

**Comment by mickem** [comment]
The first release of 0.4.4. is out on github now https://github.com/mickem/nscp/releases/tag/0.4.4.2 PLEASE BEWARE this is the very very first alpha release of 0.4.4 so use with caution, a lot of testcases still fail...

**Comment by mickem** [comment]
This is resolved in 0.5.0, if not please reopen
